cloud/blockstore/libs/endpoints/endpoint_manager.cpp

```diff
@@ -642,8 +739,17 @@ NProto::TStartEndpointResponse TEndpointManager::StartEndpointImpl(
 
     auto it = Endpoints.find(socketPath);
     if (it != Endpoints.end()) {
-        const auto& startedEndpoint = *it->second.Request;
-        return AlterEndpoint(std::move(ctx), *request, startedEndpoint);
+        auto endpoint = it->second;
+        auto error = AlterEndpoint(std::move(ctx), *request, *endpoint.Request);
+        if (HasError(error)) {
+            return TErrorResponse(error);
+        }
+
+        NProto::TStartEndpointResponse response;
+        response.MutableError()->CopyFrom(error);
+        response.MutableVolume()->CopyFrom(endpoint.Volume);
```

Comment:
here you return old volume which was saved when StartEndpoint was called but there might be change in volume attributes (for example resize)  and call to RefreshEndpoint

maybe better to call GetSession and use fresh volume value from there

Reply:
I agree. Let's do it separately.